plugins/node/opentelemetry-instrumentation-mongodb/src/instrumentation.ts

```diff
@@ -408,15 +410,35 @@ export class MongoDBInstrumentation extends InstrumentationBase<
 
     // capture parameters within the query as well if enhancedDatabaseReporting is enabled.
     const commandObj = command.query ?? command.q ?? command;
-    const query =
-      this._config?.enhancedDatabaseReporting === true
-        ? commandObj
-        : Object.keys(commandObj).reduce((obj, key) => {
-            obj[key] = '?';
-            return obj;
-          }, {} as { [key: string]: unknown });
+    const dbStatementSerializer: DbStatementSerializer =
+      this._config.dbStatementSerializer ||
+      this._defaultDbStatementSerializer.bind(this);
+
+    if (typeof dbStatementSerializer === 'function') {
```

Comment:
I think you should do this checking a bit earlier for example
```
const dbStatementSerializer: DbStatementSerializer =
      typeof this._config.dbStatementSerializer === 'function' ? this._config.dbStatementSerializer :
      this._defaultDbStatementSerializer.bind(this);
```

Reply:
done